packages/grid/x-data-grid-pro/src/hooks/features/columnReorder/useGridColumnReorder.tsx

```diff
@@ -176,22 +182,19 @@ export const useGridColumnReorder = (
         }
       }
     },
-    [props.disableColumnReorder, classes.columnHeaderDragging, logger, apiRef],
   );
 
-  const handleDragEnter = React.useCallback<
-    GridEventListener<'cellDragEnter' | 'columnHeaderDragEnter'>
-  >((params, event) => {
-    event.preventDefault();
-    // Prevent drag events propagation.
-    // For more information check here https://github.com/mui/mui-x/issues/2680.
-    event.stopPropagation();
-  }, []);
+  const handleDragEnter = useEventCallback(
+    (params: GridCellParams | GridColumnHeaderParams, event: React.DragEvent) => {
```

Comment:
We can avoid typing arguments here by adding this overload for `useEventCallback` that would match the `React.useCallback` signature:
```ts
declare module '@mui/utils' {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  export function unstable_useEventCallback<T extends Function>(fn: T): T;
}
```

Then it will be easier to switch from `React.useCallback` to `useEventCallback`:
```diff
-const handleDragEnter = React.useCallback<
+const handleDragEnter = useEventCallback<
    GridEventListener<'cellDragEnter' | 'columnHeaderDragEnter'>
  >((params, event) => {
```

Meanwhile, I'll add this overload for `unstable_useEventCallback` in the core repo.